core/vm/contracts_ec.go

```diff
@@ -73,6 +83,61 @@ func RunECPrecompiledContract(evm *EVM, precompile ECPrecompiledContract, input
 	return output, suppliedGas, err
 }
 
+// bw6761PlonkProofVerifyPrecompile implements the verification of a given PlonK proof. It is assumed that the proof
+// was generated using BW6-761 curve.
+//
+// The input is assumed to encode, in the following order:
+// - a PlonK proof of a circuit,
+// - a verifying key for the proof,
+// - a public witness suitable for verifying the proof.
```

Comment:
It is probably worth specifying a bit more about the encoding details. It's all well and good to have example code that generates it, but I would hope to see a byte-level description here.

Reply:
The thing is we don't know too much about a byte-level contents. That code fully relies on Gnark putting stuff into a buffer and getting stuff out of a buffer and every step of that process depends on Gnark. The order is all we have control over. Also, whatever Gnark puts into the buffer, is already known to be unstable: I learned the hard way that a witness serialized by Gnark v0.9.2 is not recoverable by v0.9.1. If Gnark v.0.9.3 changes the alignment, order of fields or whatever, all we can do is to update Gnark used by Quorum.

Reply:
In that case I would be extremely clear about this, and how the Gnark version used by the proving service _must be the same_ as the one used by Quorum.

Reply:
I added a paragraph about that